itest/rust/src/derive_variant.rs

```diff
@@ -0,0 +1,141 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::itest;
+use godot::bind::FromVariant;
+use godot::bind::ToVariant;
+use godot::builtin::{FromVariant, ToVariant};
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructUnit;
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructNewType(i32);
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructTuple(i32, i32);
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructNamed {
+    field1: i32,
+    field2: i32,
+}
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructGenWhere<T>(T)
+where
+    T: ToVariant + FromVariant;
+
+trait Bound {}
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+struct StructGenBound<T: Bound + ToVariant + FromVariant>(T);
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+enum Unhabitable {}
+
+#[derive(Debug, PartialEq, ToVariant, FromVariant)]
+enum Enum {
+    Variant1,
+    Variant2(i32),
+    Variant3 { data: i32 },
+    Variant4(i32, i32),
+}
```

Comment:
Maybe also here, consider naming the variants `Unit`, `OneTuple`, `Named` and `MultiTuple` or so.
Also avoids confusion with our own type `Variant`.